File: java-soap/04. CLIWEB/TP_SOAP_JAVA_CLIWEB_G5/src/main/java/ec/edu/monster/util/FormatUtil.java
package ec.edu.monster.util;

import java.util.HashMap;
import java.util.Map;

public class FormatUtil {

    private static final Map<String, String> ACCENT_MAP = new HashMap<>();

    static {
        // Países y Equipos
        ACCENT_MAP.put("Canada", "Canadá");
        ACCENT_MAP.put("Mexico", "México");
        ACCENT_MAP.put("Haiti", "Haití");
        ACCENT_MAP.put("Sudafrica", "Sudáfrica");
        ACCENT_MAP.put("Espana", "España");
        ACCENT_MAP.put("Belgica", "Bélgica");
        ACCENT_MAP.put("Iran", "Irán");
        ACCENT_MAP.put("Japon", "Japón");
        ACCENT_MAP.put("Tunez", "Túnez");
        ACCENT_MAP.put("Uzbekistan", "Uzbekistán");
        ACCENT_MAP.put("Paises Bajos", "Países Bajos");
        ACCENT_MAP.put("Corea del Sur", "Corea del Sur");
        ACCENT_MAP.put("Repechaje 1", "Repechaje 1");
        ACCENT_MAP.put("Repechaje 2", "Repechaje 2");
        ACCENT_MAP.put("UEFA A", "UEFA Stand A");
        ACCENT_MAP.put("UEFA B", "UEFA Stand B");
        ACCENT_MAP.put("UEFA C", "UEFA Stand C");
        ACCENT_MAP.put("UEFA D", "UEFA Stand D");

        // Estadios
        ACCENT_MAP.put("Estadio Ciudad de Mexico", "Estadio Ciudad de México");
        ACCENT_MAP.put("Estadio Monterrey", "Estadio BBVA Monterrey");
        ACCENT_MAP.put("Estadio Guadalajara", "Estadio Akron Guadalajara");
        ACCENT_MAP.put("Toronto Stadium", "BMO Field Toronto");
        ACCENT_MAP.put("BC Place Vancouver", "BC Place Vancouver");
        ACCENT_MAP.put("Los Angeles Stadium", "SoFi Stadium Los Ángeles");
        ACCENT_MAP.put("San Francisco Area Stadium", "Levi's Stadium San Francisco");
        ACCENT_MAP.put("Nueva Jersey Stadium", "MetLife Stadium Nueva Jersey");
        ACCENT_MAP.put("Boston Stadium", "Gillette Stadium Boston");
        ACCENT_MAP.put("Houston Stadium", "NRG Stadium Houston");
        ACCENT_MAP.put("Dallas Stadium", "AT&T Stadium Dallas");
        ACCENT_MAP.put("Philadelphia Stadium", "Lincoln Financial Field Filadelfia");
        ACCENT_MAP.put("Atlanta Stadium", "Mercedes-Benz Stadium Atlanta");
        ACCENT_MAP.put("Seattle Stadium", "Lumen Field Seattle");
        ACCENT_MAP.put("Miami Stadium", "Hard Rock Stadium Miami");
        ACCENT_MAP.put("Kansas City Stadium", "Arrowhead Stadium Kansas City");

        // Secciones
        ACCENT_MAP.put("PALCO", "Palco");
        ACCENT_MAP.put("TRIBUNA", "Tribuna");
        ACCENT_MAP.put("GENERAL", "General");
    }

    /**
     * Retorna el texto formateado con acentos o el mismo texto si no se encuentra mapeado.
     */
    public static String tilde(String text) {
        if (text == null) return "";
        String trimmed = text.trim();
        if (ACCENT_MAP.containsKey(trimmed)) {
            return ACCENT_MAP.get(trimmed);
        }
        
        // Búsqueda parcial / subcadenas (por si viene concatenado o en reportes)
        String result = text;
        for (Map.Entry<String, String> entry : ACCENT_MAP.entrySet()) {
            if (result.contains(entry.getKey())) {
                result = result.replace(entry.getKey(), entry.getValue());
            }
        }
        return result;
    }
}

File: java-soap/04. CLIMOV/TP_SOAP_JAVA_CLIMOV_G5/app/src/main/java/ec/edu/monster/servicio/FormatUtil.kt
package ec.edu.monster.servicio

object FormatUtil {

    private val accentMap = mapOf(
        "Canada" to "Canadá",
        "Mexico" to "México",
        "Haiti" to "Haití",
        "Sudafrica" to "Sudáfrica",
        "Espana" to "España",
        "Belgica" to "Bélgica",
        "Iran" to "Irán",
        "Japon" to "Japón",
        "Tunez" to "Túnez",
        "Uzbekistan" to "Uzbekistán",
        "Paises Bajos" to "Países Bajos",
        "Corea del Sur" to "Corea del Sur",
        
        "Estadio Ciudad de Mexico" to "Estadio Ciudad de México",
        "Estadio Monterrey" to "Estadio BBVA Monterrey",
        "Estadio Guadalajara" to "Estadio Akron Guadalajara",
        "Toronto Stadium" to "BMO Field Toronto",
        "BC Place Vancouver" to "BC Place Vancouver",
        "Los Angeles Stadium" to "SoFi Stadium Los Ángeles",
        "San Francisco Area Stadium" to "Levi's Stadium San Francisco",
        "Nueva Jersey Stadium" to "MetLife Stadium Nueva Jersey",
        "Boston Stadium" to "Gillette Stadium Boston",
        "Houston Stadium" to "NRG Stadium Houston",
        "Dallas Stadium" to "AT&T Stadium Dallas",
        "Philadelphia Stadium" to "Lincoln Financial Field Filadelfia",
        "Atlanta Stadium" to "Mercedes-Benz Stadium Atlanta",
        "Seattle Stadium" to "Lumen Field Seattle",
        "Miami Stadium" to "Hard Rock Stadium Miami",
        "Kansas City Stadium" to "Arrowhead Stadium Kansas City",

        "PALCO" to "Palco",
        "TRIBUNA" to "Tribuna",
        "GENERAL" to "General"
    )

    fun tilde(text: String?): String {
        if (text == null) return ""
        val trimmed = text.trim()
        if (accentMap.containsKey(trimmed)) {
            return accentMap[trimmed]!!
        }

        var result = text!!
        for ((key, value) in accentMap) {
            if (result.contains(key)) {
                result = result.replace(key, value)
            }
        }
        return result
    }
}
